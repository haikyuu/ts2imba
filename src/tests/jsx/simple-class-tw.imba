import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
function App(){
	return (
		<div className="bg-gray-200">
			<span>Test</span>
		</div>
	)
}
'''

test 'simple class' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code2 = '''
function App(){
	return (
		<div className="md:bg-gray-200">
			<span>Test</span>
		</div>
	)
}
'''

test 'simple class with breakpoint' do
	const result = await build tsx-code2
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code3 = '''
function App(){
	return (
		<div className="md:bg-gray-50">
			<span>Test</span>
		</div>
	)
}
'''

test 'falls back to color1 for color-0.5 in tailwind' do
	const result = await build tsx-code3
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')