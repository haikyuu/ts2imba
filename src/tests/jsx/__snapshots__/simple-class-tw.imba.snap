// Vitest Snapshot v1

exports[`falls back to color1 for color-0.5 in tailwind 1`] = `
"tag App
	<self>
		<div [bgc@md:gray1]>
			<span>
				\\"Test\\"
"
`;

exports[`simple class 1`] = `
"tag App
	<self>
		<div [bgc:gray2]>
			<span>
				\\"Test\\"
"
`;

exports[`simple class with breakpoint 1`] = `
"tag App
	<self>
		<div [bgc@md:gray2]>
			<span>
				\\"Test\\"
"
`;
